src/main/java/com/emc/ecs/servicebroker/service/EcsService.java

```diff
@@ -57,6 +59,11 @@ public class EcsService implements StorageService {
 
     private String objectEndpoint;
 
+    @Autowired
+    ApplicationContext applicationContext;
+
+    private S3Client s3Client; // Can not use S3Service here as it depends on this class (EcsService)
```

Comment:
this can simply be autowired:
```
    @Autowired
    private S3Client s3Client;
```
(see ObjectstoreService.java:39)

Reply:
The S3Client can not be autowired in EcsService.

The s3 secretKey is fetched in `PostConstruct` method of EcsService bean i.e. after its instance is created.
If we autowire S3Client, it will have to be initialized before EcService i.e. before calling `PostConstruct` on EcsService instance.
That will result in S3Client instance being created with `null` secretKey, which will be invalid s3 config.
To make sure that we create S3Client with correct secretKey, it needs to be created _after_ EcsService instance is created.

This is why bucketWipeOperations (which requires a valid S3Cient) is also not autowired. It is created after getting secreteKey (and EcsService instace creation).